RIOT-OS/RIOT
# sys/event/callback: in init set list_node.next to NULL

### Contribution description

When using static initializers uninitialized fields are set to 0, or
NULL for pointers. But when using event_callback_init() the
event_callback_t structure may hold non 0 values. This will lead
to the event never being called since if super.list_node.next is
not NULL it is considered as already in the event queue and therefore
is not touched.

### Testing procedure

To expose this ae1fd5260c894ab425602615f3ec5b00e6c3ef58 changes one of the event_callbacks to be initialized at runtime, and the callback never triggers, the latter commit fixes it again 6783f5509c1b378cac3ce5415c4c18ea93576b6.

On the first commit:

```
main(): This is RIOT! (Version: 2022.04-devel-268-gae1fd-HEAD)
[START] event test application.

initializing detached event queues 0x805aebc
posting 0x805409c to delayed queue at index 1
posting 0x8054094 to delayed queue at index 1
posting 0x805408c to delayed queue at index 0
running thread that will claim event queues 0x805aebc
claiming event queues 0x805aebc
launching event queue for queues 0x805aebc
triggered delayed event 0x805408c
triggered delayed event 0x805409c
triggered delayed event 0x8054094
posting 0x080540ac
posting 0x080540a4
canceling 0x080540a4
posting custom event
posting timed callback with timeout 1sec
posting timed callback with timeout 0.5sec and canceling it again
launching event queue
triggered 0x080540ac
triggered custom event with text: "CUSTOM CALLBACK"
Timeout in expect script at "child.expect_exact(u"[SUCCESS]")" (tests/events/tests/01-run.py:15)
```
